backend/dal/Helpers/Extensions/PropertyExtensions.cs

```diff
@@ -114,11 +114,23 @@ public static IQueryable<Entity.Views.Property> GenerateQuery(this PimsContext c
             if (filter.MaxAssessedValue.HasValue)
                 query = query.Where(p => p.Assessed <= filter.MaxAssessedValue);
 
+            if (filter.InEnhancedReferralProcess.HasValue && filter.InEnhancedReferralProcess.Value)
+            {
+                var statuses = context.Workflows.Where(w => w.Code.Equals("ERP"))
```

Comment:
Not sure if it will impact the SQL generated, but you really just want the `FirstOrDefault(w => w.Code == "ERP")`.  I also don't know if LINQ will translate `w.Code.Equals("ERP")`.  It will most likely request all data and put it into memory again.

Reply:
I also think this will result in pulling back properties that are in terminal statuses (i.e. Cancelled, Transfered, etc.).  You probably need to also filter on `!w.IsTerminal`.